test/core/runtime/wasm_test.cpp

```diff
@@ -0,0 +1,170 @@
+                               trie_storage,
+                               nullptr,
+                               nullptr,
+                               compile ? E::Compiled : E::Interpreted);
+  }
+  auto interpreter = [] { return make(false); };
+  auto compiler = [] { return make(true); };
+}  // namespace kagome::runtime::wasm_edge
+
+std::unique_ptr<wabt::Module> wat_to_module(std::span<const uint8_t> wat) {
+  wabt::Result result;
+  wabt::Errors errors;
+  std::unique_ptr<wabt::WastLexer> lexer =
+      wabt::WastLexer::CreateBufferLexer("", wat.data(), wat.size(), &errors);
+  if (Failed(result)) {
+    throw std::runtime_error{"Failed to parse WAT"};
+  }
+
+  std::unique_ptr<wabt::Module> module;
+  wabt::WastParseOptions parse_wast_options{{}};
+  result =
+      wabt::ParseWatModule(lexer.get(), &module, &errors, &parse_wast_options);
+  if (Failed(result)) {
+    throw std::runtime_error{"Failed to parse module"};
+  }
+  return module;
+}
+
+std::vector<uint8_t> wat_to_wasm(std::span<const uint8_t> wat) {
+  auto module = wat_to_module(wat);
+  wabt::MemoryStream stream;
+  if (wabt::Failed(wabt::WriteBinaryModule(
+          &stream,
+          module.get(),
+          wabt::WriteBinaryOptions{{}, true, false, true}))) {
+    throw std::runtime_error{"Failed to write binary wasm"};
+  }
+  return std::move(stream.output_buffer().data);
+}
+
+auto fromWat(std::string_view wat) {
+  return wat_to_wasm(kagome::str2byte(wat));
+}
+
+void test(std::string name,
+          const kagome::runtime::ModuleFactory &factory,
+          bool is_interpreter = false) {
+  using Instance = std::shared_ptr<kagome::runtime::ModuleInstance>;
+  auto read = [&](Instance instance) {
+    return fmt::format("{:x}",
+                       instance->getEnvironment()
+                           .memory_provider.get()
+                           ->getCurrentMemory()
+                           .value()
+                           .get()
+                           .view(0, 4)
+                           .value());
+  };
+  auto compile = [&](bool bulk) -> Instance {
+    auto path = fmt::format("test-wasm-{}-{}", name, bulk);
+    auto context_params = kagome::runtime::RuntimeContext::ContextParams();
+    context_params.wasm_ext_bulk_memory = bulk;
+    if (not is_interpreter) {
+      auto wat_path = kagome::filesystem::absolute(__FILE__).parent_path().string()
+                    + "/wat/memory_fill.wat";
+      std::string wat_code;
+      EXPECT_TRUE(qtils::readFile(wat_code, wat_path));
+      auto _compile = factory.compile(path, fromWat(wat_code), context_params);
+      if (not _compile) {
+        fmt::println("compile: [{}]", _compile.error().message());
+      }
+      if (bulk) {
+        EXPECT_TRUE(_compile);
+      } else {
+        EXPECT_FALSE(_compile);
+        return nullptr;
+      }
+    } else {
+      path = kagome::filesystem::absolute(__FILE__).parent_path().string()
+           + "/wasm/memory_fill.wasm";
```

Comment:
`absolute(__FILE__)` won't work
```suggestion
      path = kagome::filesystem::path{PROJECT_SOURCE_DIR} / "test/core/runtime/wat/memory_fill.wat";
```